src/main/java/com/dougnoel/sentinel/steps/TextVerificationSteps.java

```diff
@@ -100,6 +100,36 @@ public static void verifyElementTextContains(String elementName, String assertio
         }
     }
     
+    /**
+     * Waits until we can verify that an element contains certain text. It uses the text
+     * contained in double quotes for matching. If the condition is not true, it will wait 
+     * until the timeout for it to become true, such as if the text on an element changes.
+     * <b>Gherkin Examples:</b>
+     * <ul>
+     * <li>I wait until the header div contains the text "Header"</li>
+     * </ul>
+     * 
+     * @param elementName String The name of the element to be evaluated as defined in the page object.
+     * @param text String The text to verify exists in the element.
+     */
+    @Then("^I wait until the (.*?) contains the text \"([^\"]*)\"$")
+    public static void waitUntilElementTextContains(String elementName, String text) {
+    	Boolean found = getElement(elementName).waitForText(text);	
+        String elementText = getElement(elementName).getText();
+        	
+        var expectedResult = SentinelStringUtils.format(
+        		"Expected the {} element to contain the text {}. The element contained the text: {}",
+                 elementName, text, elementText
+                 	.replace("\n", " "));
+            
+        log.trace(expectedResult);
+        	
+        if (found)
+        	assertTrue(expectedResult, elementText.contains(text));
+        else
+        	assertFalse(expectedResult, elementText.contains(text));
```

Comment:
This looks like it will make it pass whether it finds the text or not. This is currently what you're going for I believe.

```suggestion
       	assertTrue(expectedResult, found);
```

Reply:
Updated to add negative case